src/content/blog/2025-01-02-python-shorts-pivoting-in-pandas.mdx

```diff
@@ -0,0 +1,184 @@
+Long datasets solve this challenge by just requiring you to add a new row with a different categorical value. However, if your data are spatial, you end up duplicating the features. You may also be limited by the value type of your observation column. Pandas can handle a column of mixed ints, floats, and strings, but most database and GIS software expect a single data type per column.
+
+## Loading the Housing Unit Inventory Dataset
+
+First, we'll use the [ArcGIS API for Python](https://developers.arcgis.com/python/latest/api-reference/) to load the ArcGIS Online copy of the HUI as a pandas dataframe. This is a large dataset with a lot of vertices, so it can take a couple minutes to load.
+
+```python
+gis = arcgis.gis.GIS()
+hui_item = gis.content.get("bbdc62867c2340949f530a71666223ab")
+hui_feature_layer = hui_item.layers[0]
+hui_dataframe = hui_feature_layer.query(return_geometry=False, as_df=True)
+```
+
+The HUI is a large dataset with polygon features for individual properties or groups of related properties (PUDs, townhomes, etc). It contains a couple categorical fields like unit type and subtype, city/county, and decade built. It also includes numerical data for each feature like unit count, acreage, and total assessed value.
+
+Like many GIS datasets, it can be considered a wide dataset that also includes categorical fields that allow us to group records together and calculate aggregate statistics from the numerical data, like averages or sums across the groups.
+
+## Using `groupby` to Summarize by Categorical Fields
+
+One way to investigate a dataset that has categorical "grouping" columns like this is to use Pandas' [`groupby`](https://pandas.pydata.org/pandas-docs/stable/user_guide/groupby.html) functionality. To figure out how the kind of housing built has changed over the years, we'll sum the unit count in each subtype per decade:
+
+```python
+hui_dataframe.groupby(["BLT_DECADE", "SUBTYPE"])["UNIT_COUNT"].sum()
+```
+
+This gives us a hierarchically-indexed series that displays the unit counts broken down by decade and subtype. This series of unit counts will be the dataset we manipulate in the next parts of this post.
+
+```
+BLT_DECADE  SUBTYPE
+1840        apartment                4.0
+            single_family            2.0
+1850        apartment                6.0
+            single_family           21.0
+1860        apartment               23.0
+                                  ...
+2020        mobile_home_park        21.0
+            single_family        42714.0
+            single_family_adu       10.0
+            townhome             20175.0
+            townhomes               15.0
+```
+
+However, if we try to plot the series out, the x-axis uses each unique combination from the MultiIndex and it just becomes a mess:
+
+```python
+hui_dataframe.groupby(["BLT_DECADE", "SUBTYPE"])["UNIT_COUNT"].sum().plot(kind="bar", figsize=(10, 6))
+```
+
+<Image src={UnstackedPlot} loading="eager" alt="Too many axis labels." />
```

Comment:
while your alt texts are comical to me, they are meant to help folks using a screen reader. To that end, they could do a better job defining the purpose for people that can't see them. I learned a lot from https://www.nngroup.com/articles/alt-text-usability/ and I suggest reviewing the rest of the alt choices after skimming that article.